src/node/snapshotter.h

```diff
@@ -25,7 +25,8 @@ namespace ccf
     static constexpr auto max_tx_interval = std::numeric_limits<size_t>::max();
 
   private:
-    ringbuffer::WriterPtr to_host;
+    ringbuffer::AbstractWriterFactory& writer_factory;
```

Comment:
Another issue TSAN spotted - multiple threads call functions in this `Snapshotter` which may try to use `to_host`. The `WriterPtr to_host` is expected to be owned and accessed by a single thread, so that's unsafe. Instead they create a `Writer` on-demand.